### PR TITLE
Improvements in Job config, adds support to Wait options

### PR DESCRIPTION
**Pull request by tallysmartins**

### CODE REVIEW

File: lib/elixir_bench_web/router.ex [hunk 1]
```diff
@@ -38,7 +38,7 @@ defmodule ElixirBenchWeb.Router do
         assign(conn, :runner, runner)
 
       {:error, _reason} ->
-        send_resp(conn, :unauthorized, Jason.encode!(%{error: "unauthorized"}))
+        halt(conn)
```
Comment (PragTob) [inline lib/elixir_bench_web/router.ex]: why the change from unauthorized to halt?

File: test/elixir_bench/benchmarks/config_test.exs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+               elixir: ^elixir_version,
+               erlang: ^erlang_version,
+               environment: %{},
+               deps: nil
+             } = changeset.data
+
+      assert changeset.valid?
+    end
+
+    test "return error if elixir and erlang errors are not supported" do
```
Comment (PragTob) [inline test/elixir_bench/benchmarks/config_test.exs]: errrors --> versions I guess?

File: test/elixir_bench/benchmarks/config_test.exs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                     image: "mysql:latest",
+                     container_name: "mysql",
+                     wait: %{port: 3306}
+                   }
+                 ]
+               }
+             } = changeset
+    end
+
+    test "allow empty docker deps in config" do
```
Comment (PragTob) [inline test/elixir_bench/benchmarks/config_test.exs]: good edge case testing! :+1:

File: test/elixir_bench_web/views/job_view_test.exs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+             commit_sha: ^commit_sha,
+             config: %{
+               deps: %{docker: []},
+               elixir_version: ^elixir,
+               environment_variables: %{},
+               erlang_version: ^erlang
+             },
+             id: ^uuid,
+             repo_slug: "owner/name"
+           } = job_data
```
Comment (PragTob) [inline test/elixir_bench_web/views/job_view_test.exs]: isn't this already tested above in the controller test? I think if we already test the view in the controller test (which is fine imo) then we don't need to duplicated that test logic in the view code or am I missing a nuance here?

When I got controller tests I often just use view tests to test edge cases that don't matter for the controller to have them setup more easily

File: lib/elixir_bench/benchmarks/config.ex [hunk 5]
```diff
@@ -15,6 +15,10 @@ defmodule ElixirBench.Benchmarks.Config do
       embeds_many :docker, Docker, primary_key: {:image, :string, []} do
         field :container_name, :string
         field :environment, {:map, :string}, default: %{}
+
+        embeds_one :wait, Wait, primary_key: false do
```
Comment (PragTob) [inline lib/elixir_bench/benchmarks/config.ex]: I'd sort of expect `Wait` to be defined somewere but it seems to work same with `Docker` - slightly confusing though :thinking:

File: lib/elixir_bench_web/router.ex (hunk 1 above)
Comment (tallysmartins) [reply lib/elixir_bench_web/router.ex]: this `send_resp` here causes response already sent error... By `halting` we just do not propagate to the controller and give back the error instantly. BasicAuth itself has the responsibility of sending the unauthorized response when the callback fails.

File: test/elixir_bench_web/controllers/job_controller_test.exs [hunk 6]
```diff
@@ -0,0 +1,60 @@
+defmodule ElixirBenchWeb.JobControllerTest do
+  use ElixirBenchWeb.ConnCase, async: false
+
+  import ElixirBenchWeb.TestHelpers
+
+  describe "claim/2" do
+    test "respond 404 when there is no job to be claimed", context do
```
[the remark this answers is not in the document]
Comment (tallysmartins) [reply test/elixir_bench_web/controllers/job_controller_test.exs]: missed that :male_detective:

File: lib/elixir_bench/benchmarks/config.ex [hunk 7]
```diff
@@ -15,6 +15,10 @@ defmodule ElixirBench.Benchmarks.Config do
...
+          field :port, :integer
+        end
```
[the remark this answers is not in the document]
Comment (tallysmartins) [reply lib/elixir_bench/benchmarks/config.ex]: well, when the logic gets too complex we move it to other module.

File: test/elixir_bench_web/views/job_view_test.exs (hunk 4 above)
Comment (tallysmartins) [reply test/elixir_bench_web/views/job_view_test.exs]: I had difficult to decide what should I test in each level. I kinda saw it with this perspective. We have mainly two specific cases that could break the View 
1) when job does not have docker deps
2) when job has docker deps

Then I replicated the test in the controller just to ensure its calling the view and the "integration" is working. Did that because I think that have/not have deps should not be concern of the controller.

File: test/elixir_bench/benchmarks/config_test.exs (hunk 2 above)
Comment (tallysmartins) [reply test/elixir_bench/benchmarks/config_test.exs]: :100: yup

Comment (tallysmartins) [reply test/elixir_bench/benchmarks/config_test.exs]: fixed
